hoomd/hpmc/external/field.py

```diff
@@ -1,243 +1,223 @@
-                 q=0.0,
-                 symmetry=[],
-                 composite=False):
-        import numpy
-        _external.__init__(self)
-        cls = None
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            if isinstance(mc, integrate.sphere):
-                cls = _hpmc.ExternalFieldLatticeSphere
-            elif isinstance(mc, integrate.convex_polygon):
-                cls = _hpmc.ExternalFieldLatticeConvexPolygon
-            elif isinstance(mc, integrate.simple_polygon):
-                cls = _hpmc.ExternalFieldLatticeSimplePolygon
-            elif isinstance(mc, integrate.convex_polyhedron):
-                cls = _hpmc.ExternalFieldLatticeConvexPolyhedron
-            elif isinstance(mc, integrate.convex_spheropolyhedron):
-                cls = _hpmc.ExternalFieldLatticeSpheropolyhedron
-            elif isinstance(mc, integrate.ellipsoid):
-                cls = _hpmc.ExternalFieldLatticeEllipsoid
-            elif isinstance(mc, integrate.convex_spheropolygon):
-                cls = _hpmc.ExternalFieldLatticeSpheropolygon
-            elif isinstance(mc, integrate.faceted_ellipsoid):
-                cls = _hpmc.ExternalFieldLatticeFacetedEllipsoid
-            elif isinstance(mc, integrate.polyhedron):
-                cls = _hpmc.ExternalFieldLatticePolyhedron
-            elif isinstance(mc, integrate.sphinx):
-                cls = _hpmc.ExternalFieldLatticeSphinx
-            elif isinstance(mc, integrate.sphere_union):
-                cls = _hpmc.ExternalFieldLatticeSphereUnion
-            elif isinstance(mc, integrate.faceted_ellipsoid_union):
-                cls = _hpmc.ExternalFieldlatticeFacetedEllipsoidUnion
-            elif isinstance(mc, integrate.convex_spheropolyhedron_union):
-                cls = _hpmc.ExternalFieldLatticeConvexPolyhedronUnion
-            else:
-                hoomd.context.current.device.cpp_msg.error(
-                    "compute.position_lattice_field: Unsupported integrator.\n")
-                raise RuntimeError(
-                    "Error initializing compute.position_lattice_field")
-        else:
-            hoomd.context.current.device.cpp_msg.error("GPU not supported yet")
-            raise RuntimeError(
-                "Error initializing compute.position_lattice_field")
 
-        self.compute_name = "lattice_field"
-        enlist = hoomd.hpmc.data._param.ensure_list
-        self.cpp_compute = cls(hoomd.context.current.system_definition,
-                               enlist(position), float(k), enlist(orientation),
-                               float(q), enlist(symmetry))
-        hoomd.context.current.system.addCompute(self.cpp_compute,
-                                                self.compute_name)
-        if not composite:
-            mc.set_external(self)
+class Harmonic(ExternalField):
+    r"""Restrain particle positions and orientations with harmonic springs.  # noqa: E501,D415,W505
```

Comment:
```suggestion
    r"""Restrain particle positions and orientations with harmonic springs.
```
As written, the `noqa` is part of the docstring. We need to format the docstring appropriately so that it obeys the linter rules.